src/details/ArborX_DetailsTreeTraversal.hpp

```diff
@@ -97,41 +100,58 @@ struct TreeTraversal<BVH, Predicates, Callback, SpatialPredicateTag>
 
     Node const *stack[64];
     Node const **stack_ptr = stack;
-    *stack_ptr++ = nullptr;
-    Node const *node = _bvh.getRoot();
+    *stack_ptr++ = nullptr;            // push sentinel value
+    Node const *node = _bvh.getRoot(); // start with root
     do
     {
-      Node const *child_left = _bvh.getNodePtr(node->left_child);
-      Node const *child_right = _bvh.getNodePtr(node->right_child);
+      int left_child_index = node->left_child;
+      int right_child_index = node->right_child;
+
+      Node const *left_child_node = _bvh.getNodePtr(left_child_index);
+      Node const *right_child_node = _bvh.getNodePtr(right_child_index);
```

Comment:
Both left and right children's indices are inside the internal node. So, there's no extra memory load associated with accessing them. However, loading the children node's here seems inefficient if you do decide to prune. Right now, the main limitation of skipping this load is because we have `isLeaf()` method part of the node and not part of the hierarchy. Otherwise, it should be possible. I had to similarly do that when trying to split internal and leaf nodes.

In short, I'm not suggesting that you change it (unless you see an easy way), just ask to add a comment.